lib/codegen/fromcto/rust/rustvisitor.js

```diff
@@ -225,6 +233,39 @@ class RustVisitor {
             }
         }
         parameters.fileWriter.writeLine(1, ')]');
+
+        if (!ModelUtil.isPrimitiveType(field.type) && ModelUtil.isMap(field)) {
+            const mapDeclaration = field.getModelFile().getType(field.getType());
+            const mapKeyType     = mapDeclaration.getKey().getType();
+            const mapValueType   = mapDeclaration.getValue().getType();
+
+            let rustKeyType;
+            let rustValueType;
+
+            // Key
+            if (ModelUtil.isPrimitiveType(mapKeyType)) {
+                rustKeyType = this.toRustType(mapKeyType);
+            } else if (ModelUtil.isScalar(mapDeclaration.getKey())) {
+                const scalarDeclaration = mapDeclaration.getModelFile().getType(mapDeclaration.getKey().getType());
+                const scalarType = scalarDeclaration.getType();
+                rustKeyType = this.toRustType(scalarType);
+            }
+            // Value
+            if (ModelUtil.isPrimitiveType(mapValueType)) {
+                rustValueType = this.toRustType(mapValueType);
+            } else if (ModelUtil.isScalar(mapDeclaration.getValue())) {
+                const scalarDeclaration = mapDeclaration.getModelFile().getType(mapDeclaration.getValue().getType());
+                const scalarType = scalarDeclaration.getType();
+                rustValueType = this.toRustType(scalarType);
+            } else {
+                rustValueType = mapValueType;
+            }
+
+            parameters.fileWriter.writeLine(1, `pub mut ${this.toValidRustName(field.getName())}: HashMap<${rustKeyType}, ${rustValueType}> = HashMap::new();`);
```

Comment:
Since none of the other generated fields a mutable by default, I think you can keep this immutable too.

Reply:
Also, I think this should be wrapped up in an option, and you are not initializing this in the `struct` definition, it's not valid Rust.
```suggestion
            parameters.fileWriter.writeLine(1, `pub ${this.toValidRustName(field.getName())}: Option<HashMap<${rustKeyType}, ${rustValueType}>>;`);
```